SteamKit2/SteamKit2/Steam/WebAPI/SteamDirectory.cs

```diff
@@ -42,42 +53,44 @@ public static Task<IReadOnlyCollection<ServerRecord>> LoadAsync( SteamConfigurat
                 ["cellid"] = configuration.CellID.ToString( CultureInfo.InvariantCulture )
             };
 
+            if ( maxNumServers.HasValue )
+            {
+                args[ "maxcount" ] = maxNumServers.Value.ToString( CultureInfo.InvariantCulture );
+            }
+
             cancellationToken.ThrowIfCancellationRequested();
 
-            var task = directory.CallAsync( HttpMethod.Get, "GetCMList", version: 1, args: args );
-            return task.ContinueWith( t =>
+            var response = await directory.CallAsync( HttpMethod.Get, "GetCMList", version: 1, args: args ).ConfigureAwait( false );
```

Comment:
Shame we can't pass the cancellation token further. Maybe worth expanding WebAPI's `AsyncInterface` someday.